src/WebJobs.Extensions.OpenAI.CosmosDBSearch/CosmosDBSearchProvider.cs

```diff
@@ -0,0 +1,266 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+            //Find if vector index exists in vectors collection
+            using (IAsyncCursor<BsonDocument> indexCursor = client.GetDatabase(databaseName).GetCollection<BsonDocument>(collectionName).Indexes.List())
+            {
+                bool vectorIndexExists = indexCursor.ToList().Any(x => x["name"] == "vectorSearchIndex");
+                if (!vectorIndexExists)
+                {
+                    BsonDocumentCommand<BsonDocument> command = new BsonDocumentCommand<BsonDocument>(
+                        vectorIndexDefinition
+                    );
+
+                    BsonDocument result = client.GetDatabase(databaseName).RunCommand(command);
+                    if (result["ok"] != 1)
+                    {
+                        this.logger.LogError("CreateIndex failed with response: " + result.ToJson());
+                    }
+                }
+            }
+            
+        }
+        catch (MongoException ex)
+        {
+            this.logger.LogError("MongoDbService CreateVectorIndexIfNotExists: " + ex.Message);
+            throw;
+        }
+
+    }
+
+    async Task UpsertVectorAsync(MongoClient client, SearchableDocument document, string collectionName)
+    {
+        List<BsonDocument> list = new List<BsonDocument>();
+        for (int i = 0; i < document.Embeddings.Response.Data.Count; i++)
+        {
+            BsonDocument vectorDocument = new()
+            {
+                { "id", Guid.NewGuid().ToString("N") },
+                { "text", document.Embeddings.Request.Input![i] },
+                { "title", Path.GetFileNameWithoutExtension(document.Title) },
+                { "embedding",  new BsonArray(document.Embeddings.Response.Data[i].Embedding.ToArray().Select(e => new BsonDouble(Convert.ToDouble(e))))},
+                { "timestamp", DateTime.UtcNow }
+            };
+            list.Add(vectorDocument);
+        }
+
+        try
+        {
+            // Insert the documents into the collection
+            await client.GetDatabase(databaseName).GetCollection<BsonDocument>(collectionName).InsertManyAsync(list);
+
+            this.logger.LogInformation("""
+                        Indexed {Count} sections
+                        """,
+                        list.Count);
+        }
+        catch (MongoException ex)
+        {
+            this.logger.LogError("MongoDbService UpsertVectorAsync: " + ex.Message);
+            throw;
+        }
+
+    }
+
+    async Task<MongoClient> GetMongoClientAsync(string connectionString)
+    {
+        // TODO: Add suport for managed identity
+        if (string.IsNullOrEmpty(connectionString))
+        {
+            throw new InvalidOperationException($"""
+                No CosmosDB mongo connection string named '{connectionString}' was found.
```

Comment:
This will always say "No CosmosDB mongo connection string named '' was found.". I think you meant to put the name of the connection string here instead of the value?